### PR TITLE
Adds orientation selection

### PR DESCRIPTION
**Pull request by anmoljagetia**
@GordonSmith @mlzummo Please review.
Signed-off-by: Anmol Jagetia <anmoljagetia@gmail.com>

### CODE REVIEW

File: src/chart/XYAxis.js [hunk 1]
```diff
@@ -122,24 +125,27 @@
...
+            this.x.rangeRoundBands([0, this.orientation() === "horizontal" ? width : height], 0.1);
         } else if (this.x.rangeRound) {
-            this.x.range([0, width]);
+            this.x.range([0, this.orientation() === "horizontal" ? width : height]);
         }
-        this.y.range([height, 0]);
+        this.y.range([this.orientation() === "horizontal" ? height : 0, this.orientation() === "horizontal" ? 0 : width - margin.left - margin.right]);
 
-        //  Render  ---
+         // Render  ---
```
Comment (GordonSmith) [inline src/chart/XYAxis.js]: Typo

File: src/chart/XYAxis.js [hunk 2]
```diff
@@ -15,6 +15,8 @@
     }
     XYAxis.prototype = Object.create(SVGWidget.prototype);
 
+    XYAxis.prototype.publish("orientation", "horizontal", "set", "...", ["horizontal", "vertical"]);
```
Comment (GordonSmith) [inline src/chart/XYAxis.js]: Replace "..." with meaningful description.

File: src/chart/XYAxis.js [hunk 3]
```diff
@@ -62,13 +64,11 @@
         var test = element.append("g");
 
         var svgXAxis = test.append("g")
-            .attr("class", "x axis")
```
Comment (GordonSmith) [inline src/chart/XYAxis.js]: You will want the correct class here to get a 100% correct margin calculation.

File: src/chart/XYAxis.js [hunk 4]
```diff
@@ -62,13 +64,11 @@
...
             .attr("transform", "translate(0," + height + ")")
-            .call(this.xAxis)
+            .call(this.orientation() === "horizontal" ? this.xAxis : this.yAxis)
         ;
         var svgYAxis = test.append("g")
-            .attr("class", "y axis")
```
Comment (GordonSmith) [inline src/chart/XYAxis.js]: You will want the correct class here to get a 100% correct margin calculation

File: src/chart/Column.js [hunk 5]
```diff
@@ -34,14 +34,28 @@
...
+                .attr("x", function (d) { return context.x(d[0]); })
+                .attr("width", this.x.rangeBand())
+                .attr("y", function (d) { return context.y(d[1]); })
+                .attr("height", function (d) { return height - context.y(d[1]); })
+                .style("fill", function (d) { return context._palette(d[0]); })
+            ;
+        } else {
+            column.transition()
+                .attr("class", "columnRect")
+                // .attr("x", function (d) { return context.y(d[1]); })
```
Comment (GordonSmith) [inline src/chart/Column.js]: Remove comment code

File: src/chart/Column.js [hunk 6]
```diff
@@ -34,14 +34,28 @@
...
+                .attr("x", function (d) { return 0; })
+                .attr("height", this.x.rangeBand())
+                .attr("y", function (d) { return context.x(d[0]); })
+                // .attr("width", function (d) { return width - context.y(d[1]); })
```
Comment (GordonSmith) [inline src/chart/Column.js]: Remove comment code

File: src/chart/Line.js [hunk 7]
```diff
@@ -22,18 +22,32 @@
...
+                .y(function (d) {
+                    switch (context._xScale) {
+                        case "DATE":
+                            return context.x(context.parseDate(d[0]));
+                    }
+                    return context.x(d[0]) + (context.x.rangeBand ? context.x.rangeBand() / 2 : 0);
+                })
+                .x(function (d) { return context.y(d[1]); })
+            ;
+        }
```
Comment (GordonSmith) [inline src/chart/Line.js]: Above the two functions used in the .x and .y calls don't change so you can declare them in one place (beside the ```var d3Line```) and just refer to them inside the if/else block?

**Comment by GordonSmith** [comment]
Looks mostly ok - see comments above...

**Comment by GordonSmith** [comment]
Can you re-submit against candidate-1.2.0 and I will pull then.

**Comment by mlzummo** [comment]
looks good to me

**Comment by anmoljagetia** [comment]
Closing this in favor of #454.

**Pull request closed without merging**